pasta_fp.c

```diff
@@ -2144,12 +2144,18 @@ void fiat_pasta_fp_print(const uint64_t x[4]) {
     printf("] \n");
 }
 
-void fiat_pasta_fp_sqrt(uint64_t x[4], const uint64_t value[4]) {
+bool fiat_pasta_fp_sqrt(uint64_t x[4], const uint64_t value[4]) {
     // A few assertions to make sure s, t, and nqr are initialized.
 
     if (fiat_pasta_fp_equals_zero(value)) {
       for (size_t j = 0; j < 4; ++j) { x[j] = 0; }
-      return;
+      return true;
+    }
+
+    uint64_t check[4];
```

Comment:
shouldn't you be using the types you defined elsewhere as `typedef uint64_t Field[LIMBS_PER_FIELD];`